File: src/main/java/com/rtsup/data2/Room.java
package com.rtsup.data2;

import com.rtsup.data.RoomStatus;
import org.springframework.data.annotation.Id;

public class Room {

    @Id
    private String id;

    private String name;
    private String description;
    private String address;
    private String userLockId;
    private RoomStatus status;

    public Room() {
    }

    public Room(String name) {
        this.name = name;
    }

    public String getUserLockId() {
        return userLockId;
    }

    public void setUserLockId(String userLockId) {
        this.userLockId = userLockId;
    }

    public String getId() {
        return id;
    }

    public RoomStatus getStatus() {
        return status;
    }

    public void setStatus(RoomStatus status) {
        this.status = status;
    }

    public String getName() {
        return name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }
}

File: src/main/kotlin/com/rtsup/data/Room.kt
package com.rtsup.data

import com.fasterxml.jackson.annotation.JsonProperty
import org.springframework.data.annotation.Id

// curl -i -X POST -H "Content-Type:application/json" -d '{  "name" : "Test2",  "address" : "Test Address",  "status": "busy" }' http://localhost:8080/rooms
public class Room(
        @Id var id: String?,
        var name: String,
        var address: String?,
        var description: String?,
        var userLockId: String?,
        var status: RoomStatus) {

}

public enum class RoomStatus() {
    @JsonProperty("free")
    FREE,
    @JsonProperty("busy")
    BUSY,
    @JsonProperty("inactive")
    INACTIVE
}
